### PR TITLE
Textures, materials and shaders improvements

### PR DESCRIPTION
**Pull request by lewa-j**
Should fix https://github.com/SteamDatabase/ValveResourceFormat/issues/114 for viewer, but not sure about export.
before:
img
after:
img
before:
img
after:
img
before:
vrf exe Screenshot 2020 03 29 - 00 36 59 94
after:
vrf exe Screenshot 2020 03 28 - 17 15 12 48

### CODE REVIEW

**Comment by lewa-j** [comment]
Wait, don't merge. I realized that changes in ShaderLoader were a mistake.

**Comment by lewa-j** [comment]
Now good.

**Comment by xPaw** [comment]
hemiOctRB is likely used in DXT5 too, did you look into that?

File: GUI/Types/Renderer/Shaders/simple.frag [hunk 1]
```diff
@@ -62,8 +62,8 @@ void main()
 #else
     //Calculate half-lambert lighting
     float illumination = dot(worldNormal, lightDirection);
-    illumination = illumination * 0.5 + 0.5;
-    illumination = illumination * illumination;
+    //illumination = illumination * 0.5 + 0.5;
+    //illumination = illumination * illumination;
```
Comment (Perryvw) [inline GUI/Types/Renderer/Shaders/simple.frag]: This should be reverted, this formula is the half-lambert equation Valve is known to use.

File: GUI/Types/Renderer/Shaders/vr_standard.frag [hunk 2]
```diff
@@ -74,9 +94,10 @@ void main()
 #else
     //Calculate half-lambert lighting
     float illumination = dot(worldNormal, lightDirection);
-    illumination = illumination * 0.5 + 0.5;
-    illumination = illumination * illumination;
-    illumination = min(illumination + 0.3, 1.0);
+    //illumination = illumination * 0.5 + 0.5;
+    //illumination = illumination * illumination;
+    //illumination = min(illumination + 0.3, 1.0);
+    illumination = illumination * 0.7 + 0.3;
```
Comment (Perryvw) [inline GUI/Types/Renderer/Shaders/vr_standard.frag]: This should probably also be reverted

File: ValveResourceFormat/Resource/ResourceTypes/Material.cs [hunk 3]
```diff
@@ -93,6 +96,13 @@ public override void Read(BinaryReader reader, Resource resource)
                 arguments.Add(name, value != 0);
             }
 
+            var specialDeps = (SpecialDependencies)Resource.EditInfo.Structs[ResourceEditInfo.REDIStruct.SpecialDependencies];
```
Comment (Perryvw) [inline ValveResourceFormat/Resource/ResourceTypes/Material.cs]: do we really have to use editinfo for this? Is there not some flag in the material DATA that we can use instead (they are automatically loaded into the shader)

**Comment by lewa-j** [comment]
I don't found any dxt5 normalmaps with it. Do you know examples in hla or steamvr?

File: GUI/Types/Renderer/Shaders/simple.frag [hunk 4]
```diff
@@ -86,14 +86,14 @@ void main()
 #endif
 
 #if param_renderMode_Normals
-	outputColor = vec4(vNormalOut, 1.0);
+	outputColor = vec4(vNormalOut * vec3(0.5) + vec3(0.5), 1.0);
```
Comment (Perryvw) [inline GUI/Types/Renderer/Shaders/simple.frag]: All fragment shaders should visualize normals and bump normals the same. So either revert this change or update the renderMode code in the other fragment shaders too. I personally prefer keeping it the way it was.

File: GUI/Types/Renderer/Shaders/simple.frag (hunk 1 above)
Comment (lewa-j) [reply GUI/Types/Renderer/Shaders/simple.frag]: I found mentions only about Source 1. Source 2 should have a more physically accurate lighting model.

File: ValveResourceFormat/Resource/ResourceTypes/Material.cs (hunk 3 above)
Comment (lewa-j) [reply ValveResourceFormat/Resource/ResourceTypes/Material.cs]: I don't see any. At least from that VRF shows.

File: GUI/Types/Renderer/Shaders/simple.frag (hunk 4 above)
Comment (lewa-j) [reply GUI/Types/Renderer/Shaders/simple.frag]: As it was before, all negative values would be just black. It half the spectrum. I'll update the rest.

File: GUI/Types/Renderer/Shaders/simple.frag (hunk 1 above)
Comment (Perryvw) [reply GUI/Types/Renderer/Shaders/simple.frag]: Unless we find explicit references for this we assume Source 2 is doing the same as Source 1.

Comment (lewa-j) [reply GUI/Types/Renderer/Shaders/simple.frag]: http://media.steampowered.com/apps/valve/2015/Alex_Vlachos_Advanced_VR_Rendering_GDC2015.pdf
page 42

Comment (Perryvw) [reply GUI/Types/Renderer/Shaders/simple.frag]: Very interesting, in that case can you remove the old commented code?

Comment (lewa-j) [reply GUI/Types/Renderer/Shaders/simple.frag]: Sure
